ARTstor.js

```diff
@@ -1,131 +1,899 @@
+        - check if the page is image viewer
+            - get the object from the image viewer, then process the id.
+    - Process the id
+        - find the object type.
+        - get the metadta service url from id using service call  http://library.artstor.org/library/secure/metadata/id
+            - fetch and convert the metadata from the metadata service call
+                - take into consideration of different metadata field for the portals
+                - may need to convert/format the data values.
+            - fetch the item notes using: 
+        - get the resource link url from id: :http://library.artstor.org/library/secure/metadata/id?_method=FpHtml
+            - fetch the resource from resource url
+            - set the item title and item mine type.
 
-	if (getData(doc, true)) {
-		return "artwork";
-	}
+    doWeb is run when a user, wishing to save one or more items, activates the selected translator. 
+    Sidestepping the retrieval of item metadata, we'll first focus on how doWeb can be used to save 
+    retrieved item metadata (as well as attachments and notes) to your Zotero library.**/
+function doWeb(doc, url) {
+    if (url.match(/\/iv2/)) {
+        doImageViewer(doc, url);
+    }
+    if (url.match(/#3\|/)) {
+        // Thumbnail window page
+        if ((doc.getElementsByClassName('MetaDataWidgetRoot') != null) && (doc.getElementsByClassName('MetaDataWidgetRoot').length > 0)) {
+            doMetadataWindow(doc, url);
+        } else {
+            var skipThumbnail = false;
+            if ((doc.getElementById("floatingPlaceHolder") != null) && (doc.getElementById("floatingPlaceHolder").style.display == "block")) {
+                // Don't capture date if small window is present
+                skipThumbnail = true;
+            }
+            if ((doc.getElementById("thumbNavSave1") != null) && (doc.getElementById("thumbNavSave1").style.display == "block")) {
+                // Don't capture data if image group window is in editing state.
+                skipThumbnail = true;
+            }
+            if ((doc.getElementById("ssContentWrap") != null) && (doc.getElementById("ssContentWrap").style.display == "inline")) {
+                // Don't capture data if slide show window is present
+                skipThumbnail = true;
+            }
+            // Allow thumbnail window.
+            if (!skipThumbnail) {
+                doThumbnails(doc, url);
+            }
+        }
+    }
 }
 
-function associateData(newItem, value, field) {
-	if (!value) return;
-	if (typeof value == 'string') value = ZU.trimInternal(value);
-	if (!value) return;
-	
-	newItem[field] = value;
+function doImageViewer(doc, url) {
+    // get the image id and object type from the page
+    // this contains the objId and object type separate by : as in "AWSS35953_35953_25701160:11"
+    var objID = doc.getElementById("objID");
+    if (objID != null) {
```

Comment:
You should guarantee that this is present in `detectWeb`. It's better to let `doWeb` throw if it's not able to import something, since we then get a proper error report.